### PR TITLE
Remove functionality for rendering collision models directly in the Model class

### PR DESCRIPTION
**Pull request by sabrina-shanman**
https://highfidelity.manuscript.com/f/cases/12114/Draw-Collision-Shapes-causes-affected-entities-to-simply-disappear
From the end-user's standpoint, this PR results in the menu option "Draw Collision Shapes" being removed from the Physics section of the developer menu.

Behind the scenes, the Model class no longer serves the dual purpose of rendering aesthetic models and physics models. The physics model rendering functionality will later be re-implemented elsewhere. Some of the generic logic for switching between these two types of models is still present in RenderableModelEntityItem, for later implementation.

## TEST PLAN
- In this PR, check that the "Draw Collision Shapes" option in the Physics section of the developer menu is not present.

### CODE REVIEW

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10521/HighFidelity-Beta-PR13350-4c39d08.apk

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10521/HighFidelity-Beta-PR13350-4c39d08.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10521/HighFidelity-Beta-Interface-PR13350-4c39d08.exe

**Comment by sabrina-shanman** [comment]
An unused constant was blocking the Mac build. The constant has been removed so it should be fixed now.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10567/HighFidelity-Beta-PR13350-82fddfc.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10567/HighFidelity-Beta-PR13350-82fddfc.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10567/HighFidelity-Beta-Interface-PR13350-82fddfc.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10567/HighFidelity-Beta-PR13350-82fddfc.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10567/HighFidelity-Beta-Interface-PR13350-82fddfc.exe

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 1]
```diff
@@ -1052,6 +1052,7 @@ using namespace render;
 using namespace render::entities;
 
 ModelEntityRenderer::ModelEntityRenderer(const EntityItemPointer& entity) : Parent(entity) {
+    // TODO: Re-add menu item in Menu.ccp/Menu.h when collision model updates in doRenderUpdateSynchronousTyped(...) are re-implemented
     connect(DependencyManager::get<EntityTreeRenderer>().data(), &EntityTreeRenderer::setRenderDebugHulls, this, [&] {
         _needsCollisionGeometryUpdate = true;
         emit requestRenderUpdate();
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Purge the above `connect` and also `EntityTreeRenderer::setRenderDebugHulls()`.

File: libraries/entities-renderer/src/RenderableModelEntityItem.cpp [hunk 2]
```diff
@@ -1416,19 +1417,18 @@ void ModelEntityRenderer::doRenderUpdateSynchronousTyped(const ScenePointer& sce
...
                 // release mesh
                 collisionMeshCache.releaseMesh(_collisionMeshKey);
             }
-            // clear model's collision geometry
-            graphics::MeshPointer mesh = nullptr;
-            _model->setCollisionMesh(mesh);
+            // TODO: Stop displaying collision model
+
         }
     }
```
Comment (AndrewMeadows) [inline libraries/entities-renderer/src/RenderableModelEntityItem.cpp]: Purge this entire `if` block I think.  Which means you can purge `_needsCollisionGeometryUpdate` and also `collisionMeshCache`

File: libraries/render-utils/src/Model.cpp [hunk 3]
```diff
@@ -777,11 +725,14 @@ void Model::updateRenderItemsKey(const render::ScenePointer& scene) {
...
         });
     }
+    //TODO: Remove
+#ifdef RENDER_MODEL_BOUNDING_BOXES
     foreach(auto item, _collisionRenderItemsMap.keys()) {
         transaction.updateItem<ModelMeshPartPayload>(item, [renderItemsKey](ModelMeshPartPayload& data) {
             data.updateKey(renderItemsKey);
         });
     }
+#endif
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/Model.cpp]: Better to just remove it now than to flag it for later removal.

File: libraries/render-utils/src/Model.h [hunk 4]
```diff
@@ -430,9 +426,7 @@ public slots:
     QVector<TriangleSet> _modelSpaceMeshTriangleSets; // model space triangles for all sub meshes
 
 
-    void createRenderItemSet();
     virtual void createVisibleRenderItemSet();
```
Comment (AndrewMeadows) [inline libraries/render-utils/src/Model.h]: Maybe change the method's name:  `createVisibleRenderItemSet()` --> `createRenderItemSet()`.

**Comment by hifi-gustavo** [comment]
Android build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dandroid/10595/HighFidelity-Beta-PR13350-79fd3cb.apk

**Comment by hifi-gustavo** [comment]
Mac OS X build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10595/HighFidelity-Beta-PR13350-79fd3cb.dmg

**Comment by hifi-gustavo** [comment]
Mac OS X client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dmacos/10595/HighFidelity-Beta-Interface-PR13350-79fd3cb.dmg

**Comment by hifi-gustavo** [comment]
Windows build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10595/HighFidelity-Beta-PR13350-79fd3cb.exe

**Comment by hifi-gustavo** [comment]
Windows client-only build is available at: https://deployment.highfidelity.com/jobs/pr-build/label%3Dwindows/10595/HighFidelity-Beta-Interface-PR13350-79fd3cb.exe

**Comment by QliCknox** [comment]
Currently Testing

**Pull request merged**